### ISSUE TITLE
Problems installing dplyover

### ISSUE THREAD

**Issue by TimTeaFan** [TimTeaFan/dplyover#10]
A user on SO reported problems installing {dplyover}. The following error was thrown:

`Error: Failed to install 'dplyover' from GitHub: 'local_makevars' is not an exported object from 'namespace:withr'`.

**Comment by realkrantz** [comment]
R version 4.1.0 (2021-05-18)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] grid      parallel  stats     graphics  grDevices utils    
[7] datasets  methods   base     

other attached packages:
 [1] utf8_1.2.1           magrittr_2.0.1      
 [3] officer_0.3.18       rgl_0.106.8         
 [5] brms_2.15.0          Rcpp_1.0.6          
 [7] caret_6.0-88         lattice_0.20-44     
 [9] factoextra_1.0.7     VIM_6.1.0           
[11] colorspace_2.0-1     prettyR_2.2-3       
[13] forcats_0.5.1        dplyr_1.0.6         
[15] purrr_0.3.4          readr_1.4.0         
[17] tidyr_1.1.3          tibble_3.1.2        
[19] tidyverse_1.3.1      stringr_1.4.0       
[21] reshape2_1.4.4       cowplot_1.1.1       
[23] ggpubr_0.4.0         reshape_0.8.8       
[25] doParallel_1.0.16    iterators_1.0.13    
[27] foreach_1.5.1        rstan_2.21.2        
[29] ggplot2_3.3.3        StanHeaders_2.21.0-7

loaded via a namespace (and not attached):
  [1] uuid_0.1-4              readxl_1.3.1           
  [3] backports_1.2.1         igraph_1.2.6           
  [5] plyr_1.8.6              sp_1.4-5               
  [7] splines_4.1.0           crosstalk_1.1.1        
  [9] rstantools_2.1.1        inline_0.3.18          
 [11] digest_0.6.27           htmltools_0.5.1.1      
 [13] rsconnect_0.8.18        fansi_0.4.2            
 [15] remotes_2.3.0           openxlsx_4.2.3         
 [17] recipes_0.1.16          modelr_0.1.8           
 [19] gower_0.2.2             RcppParallel_5.1.4     
 [21] matrixStats_0.58.0      xts_0.12.1             
 [23] prettyunits_1.1.1       rvest_1.0.0            
 [25] ggrepel_0.9.1           xfun_0.23              
 [27] haven_2.4.1             callr_3.7.0            
 [29] crayon_1.4.1            jsonlite_1.7.2         
 [31] lme4_1.1-27             survival_3.2-11        
 [33] zoo_1.8-9               glue_1.4.2             
 [35] gtable_0.3.0            ipred_0.9-11           
 [37] V8_3.4.2                car_3.0-10             
 [39] pkgbuild_1.2.0          DEoptimR_1.0-9         
 [41] abind_1.4-5             scales_1.1.1           
 [43] mvtnorm_1.1-1           DBI_1.1.1              
 [45] rstatix_0.7.0           miniUI_0.1.1.1         
 [47] xtable_1.8-4            laeken_0.5.1           
 [49] foreign_0.8-81          proxy_0.4-25           
 [51] DT_0.18                 stats4_4.1.0           
 [53] lava_1.6.9              prodlim_2019.11.13     
 [55] vcd_1.4-8               htmlwidgets_1.5.3      
 [57] httr_1.4.2              threejs_0.3.3          
 [59] ellipsis_0.3.2          pkgconfig_2.0.3        
 [61] loo_2.4.1               nnet_7.3-16            
 [63] dbplyr_2.1.1            manipulateWidget_0.11.0
 [65] later_1.2.0             tidyselect_1.1.1       
 [67] rlang_0.4.11            munsell_0.5.0          
 [69] cellranger_1.1.0        tools_4.1.0            
 [71] cli_2.5.0               generics_0.1.0         
 [73] ranger_0.12.1           broom_0.7.6            
 [75] ggridges_0.5.3          fastmap_1.1.0          
 [77] knitr_1.33              ModelMetrics_1.2.2.2   
 [79] processx_3.5.2          fs_1.5.0               
 [81] zip_2.1.1               robustbase_0.93-7      
 [83] nlme_3.1-152            mime_0.10              
 [85] projpred_2.0.2          xml2_1.3.2             
 [87] shinythemes_1.2.0       compiler_4.1.0         
 [89] bayesplot_1.8.0         rstudioapi_0.13        
 [91] curl_4.3.1              gamm4_0.2-6            
 [93] e1071_1.7-7             ggsignif_0.6.1         
 [95] reprex_2.0.0            stringi_1.6.2          
 [97] ps_1.6.0                Brobdingnag_1.2-6      
 [99] Matrix_1.3-3            markdown_1.1           
[101] nloptr_1.2.2.2          shinyjs_2.0.0          
[103] vctrs_0.3.8             pillar_1.6.1           
[105] lifecycle_1.0.0         lmtest_0.9-38          
[107] bridgesampling_1.1-2    data.table_1.14.0      
[109] httpuv_1.6.1            R6_2.5.0               
[111] promises_1.2.0.1        gridExtra_2.3          
[113] rio_0.5.26              codetools_0.2-18       
[115] gtools_3.8.2            colourpicker_1.1.0     
[117] boot_1.3-28             MASS_7.3-54            
[119] assertthat_0.2.1        rprojroot_2.0.2        
[121] withr_2.2.0             shinystan_2.5.0        
[123] mgcv_1.8-35             hms_1.1.0              
[125] rpart_4.1-15            timeDate_3043.102      
[127] coda_0.19-4             class_7.3-19           
[129] minqa_1.2.4             carData_3.0-4          
[131] pROC_1.17.0.1           base64enc_0.1-3        
[133] shiny_1.6.0             lubridate_1.7.10       
[135] dygraphs_1.1.1.6

**Comment by TimTeaFan** [comment]
Thanks for adding this information! I really wonder why {withr}  should cause problems. I'm using 2.3.0, but I'm not aware that I used `withr:: local_makevars` anywhere in the code. Even the tests should not contain it. I'll keep looking, but my guess is that you should be able to install it with withr >= 2.3.0.

**Comment by realkrantz** [comment]
Also, see this seemingly related [issue](https://github.com/stan-dev/rstan/pull/857). Because of that, I need to use `withr v. 2.2.0`, as `withr v. 2.2.0` creates problem with '[Stan](https://discourse.mc-stan.org/t/new-error-cleanup-makevar-old-argument-rmu-is-missing-with-no-default/18137/30)`.

**Comment by realkrantz** [comment]
Also, see this seemingly related [issue](https://github.com/stan-dev/rstan/pull/857). Because of that, I need to use `withr v. 2.2.0`, as `withr v. 2.2.0` creates another '[issue](https://discourse.mc-stan.org/t/new-error-cleanup-makevar-old-argument-rmu-is-missing-with-no-default/18137/30)`.

**Comment by realkrantz** [comment]
Also, see this seemingly related [issue](https://github.com/stan-dev/rstan/pull/857). Because of that, I need to use `withr v. 2.2.0`, as `withr v. 2.2.0` creates another [issue](https://discourse.mc-stan.org/t/new-error-cleanup-makevar-old-argument-rmu-is-missing-with-no-default/18137/30).

**Comment by realkrantz** [comment]
Also, see this seemingly related [issue](https://github.com/stan-dev/rstan/pull/857). Because of that, I need to use `withr v. 2.2.0`, as `withr v. 2.3.0` creates another '[issue](https://discourse.mc-stan.org/t/new-error-cleanup-makevar-old-argument-rmu-is-missing-with-no-default/18137/30)`.

**Comment by realkrantz** [comment]
Also, see this seemingly related [issue](https://github.com/stan-dev/rstan/pull/857). Because of that, I need to use `withr v. 2.2.0`, as `withr v. 2.3.0` creates another [issue](https://discourse.mc-stan.org/t/new-error-cleanup-makevar-old-argument-rmu-is-missing-with-no-default/18137/30).

**Comment by TimTeaFan** [comment]
I now installed {withr} v. 2.2.0 and I was able to replicate this error. Apparently, it has nothing to do with {dplyover} itself, since I was not able to install any packages with `remotes::install_github`. It always threw the same error. I thought downgrading {remotes} to a prior version would help, but apparently that was not the case. I'm not even use if {remotes} call `withr::local_makevars` or if it just triggers something else which then makes the call.

So if you are unable to upgrade to {withr} >= v. 2.3.0 then another option would be to download the .tar.gz file and install via `install.packages("dplyover-0.0.8.9000.tar", repos=NULL, type="source")`. This worked without problems in my case (Mac).

Since this issue does not seem to be related to {dplyover} I'm gonna close it.
